Data/Mod/Word.hs

```diff
@@ -247,14 +250,14 @@ instance KnownNat m => Fractional (Mod m) where
 -- >>> invertMod 4 :: Mod 10 -- 4 and 10 are not coprime
 -- Nothing
 invertMod :: KnownNat m => Mod m -> Maybe (Mod m)
-invertMod mx@(Mod x) = case natVal mx of
+invertMod mx@(Mod !x) = case natVal mx of
```

Comment:
`Mod` is a newtype, what's the reason to add a bang here?

Reply:
It's a newtype, so matching the constructor doesn't force the value. Adding the bang causes GHC to generate a version which takes a `Word#`, which gets used for calls to `invertMod` which previously passed a boxed `Word`.